File: 30 Days of Code/Day 20 Sorting/Solution.py

import sys

def swap( array, idx, jdx ):
    temp = array[ idx ]
    array[ idx ] = array[ jdx ]
    array[ jdx ] = temp

n = int( input( ).strip( ) )
array = list( map( int, input( ).strip( ).split( ' ' ) ) )
numberOfSwaps = 0

for idx in range( n ):
    for jdx in range( n - 1 ):
        if array[ jdx ] > array[ jdx + 1 ]:
            swap( array, jdx, jdx + 1 )
            numberOfSwaps += 1
    
    if numberOfSwaps == 0:
        break

print( "Array is sorted in " + str( numberOfSwaps ) + " swaps." )
print( "First Element: " + str( array[ 0 ] ) )
print( "Last Element: " + str( array[ n - 1 ] ) )

File: 30 Days of Code/Day 20 Sorting/Solution.java
import java.util.Scanner;

public class Solution {

    static void swap( int[ ] array, int idx, int jdx ) {
        int temp = array[ idx ];
        array[ idx ] = array[ jdx ];
        array[ jdx ] = temp;
    }

    public static void main( String[ ] args ) {
        Scanner in = new Scanner( System.in );
        int n = in.nextInt( );
        int[ ] array = new int[ n ];

        for( int idx = 0; idx < n; idx++ ) {
            array[ idx ] = in.nextInt( );
        }
        in.close( );

        for( int idx = 0; idx < n; idx++ ) {
            int numberOfSwaps = 0;

            for( int jdx = 0; jdx < n - 1; jdx++ ) {
                if( a[ jdx ] > a[ jdx + 1 ] ) {
                    swap( a, jdx, jdx + 1);
                    numberOfSwaps += 1;
                }
            }

            if( numberOfSwaps == 0 ) {
                break;
            }
        }

        System.out.println( "Array is sorted in " + numberOfSwaps + " swaps." );
        System.out.println( "First Element: " + array[ 0 ] );
        System.out.println( "Last Element: " + array[ n - 1 ] );        
    }
}